### ISSUE TITLE
Preferences tab is blank

### ISSUE THREAD

**Issue by ngirard** [piroor/treestyletab#2888]
Using TST 3.8.1 on Firefox 88.0 and Ubuntu 21.04.

The preferences tab is completely blank:

![Screenshot_20210507_065744](https://user-images.githubusercontent.com/98098/117399769-9126cf80-af01-11eb-8027-251fa6a104d5.png)

**Comment by piroor** [comment]
I've tried TST 3.8.1 on Firefox 88.0 on Ubuntu 20.04LTS and they looks working as expected. The options page is shown after a while. Could you try with a clean profile?
https://github.com/piroor/treestyletab/wiki/How-to-run-Firefox-with-a-clean-profile

**Comment by ngirard** [comment]
Hey @piroor,
it works fine within a clean profile.
I'm puzzled...

**Comment by piroor** [comment]
I think a bisection will help you to find out the trigger of the problem.
https://github.com/piroor/treestyletab/wiki/How-to-minimize-conditions-to-reproduce-a-problem

**Comment by ngirard** [comment]
Sure. I'm afraid I can't afford to do this now (GMT+2 here) ; will try to come back to it tonight.

Cheers, and many thanks for your time & assistance !

**Comment by ngirard** [comment]
Just a heads up.

Since the problem occured after I upgraded both my system *and* TST, I suspected that the problem could be related to the system upgrade.
I tried switching from X11 to Wayland and from various rendering engines (XRender vs OpenGL), but it didn't make any difference.

Also the problem affects other preferences' tab, but not all.
For instance, the preferences tab for the "Bookmark search plus 2" extension is not affected:
![Screenshot_20210508_164958](https://user-images.githubusercontent.com/98098/117543598-d3453380-b01d-11eb-8a17-7ebdad50e3be.png)

but the preferences tab for the "Copy selected tabs to clipboard extension" is:
![Screenshot_20210508_165321](https://user-images.githubusercontent.com/98098/117543696-2f0fbc80-b01e-11eb-8e21-344b2bd63515.png)

**Comment by ngirard** [comment]
Nailed it.
The following CSS rule is applied to the `body` of the preferences `#document`:
```css
:root:not(.initialized) > * {
	opacity: 0;
}
```
![Screenshot_20210508_200100](https://user-images.githubusercontent.com/98098/117549007-4fe50b80-b038-11eb-8a82-aea5afc1c569.png)

Disabling it makes the preferences to appear:
![Screenshot_20210508_200322](https://user-images.githubusercontent.com/98098/117549115-e1547d80-b038-11eb-9b47-37010a0445b0.png)

Hope this helps.

**Comment by irvinm** [comment]
@ngirard so I guess your options page was not being initialized properly.  Are you still seeing this or are you manually changing the CSS?  (Or did you simply move to a new profile which you said was working)  I'm wondering if you could close this item or if there was more to do.

**Comment by irvinm** [comment]
@ngirard can you close this item?

**Comment by ngirard** [comment]
Sorry, got overwhelmed by work.

@irvinm, this is also my interpretation that the options page isn't being properly initialized.

The screenshot I attached was the result of a manual intervention from me ; without touching anything the options page stays blank.

The defect is still present, despite upgrading TST to latest version and Firefox to v89.0.

HTH

**Comment by irvinm** [comment]
Thanks @ngirard.  Have you tried a new profile with the same versions to see if that works or not?

**Comment by ngirard** [comment]
@irvinm, it's working fine with a clean profile.

**Comment by irvinm** [comment]
It looks like things get considered initialized at this point in the process:

https://github.com/piroor/treestyletab/blob/7255652db55c376e0da116a084e3e4dd6fabc990/webextensions/options/init.js#L865

However, there is quite a few things that are done in init.js before that point.  Have you tried exporting your settings and then resetting them in you main profile to see if it works.  (You already know it works in a new profile but if it doesn't work in your main profile with reset options that it has something to do with the profile itself or a conflict with another addon in that profile)

The other thing you could also try is exporting your setting from TST and importing them into that new profile and ensure that things still work.  If that works, it would prove it wasn't any of the options you are using with TST.

Just some thoughts ...

**Comment by piroor** [comment]
I've introduced some changes 642caa9 and 546a1bf, now TST continues initialization process of the options page even if there are some failures. This is not the best solution, but I hope it should help people suffered from whilte-outed options page and they can access to available options.

**Comment by ngirard** [comment]
@piroor, thank you very much for your efforts, and apologizes for not being able to devote enough time to qualify this defect any further, despite @irvinm's wise thoughts.

I might be able to find some time after june 21st if needed, though.

**Comment by confluence** [comment]
I am experiencing this issue in 3.8.3 with Firefox 89 on Ubuntu 20.04. I can't use the workaround because I can't see the `#document` node at all in the inspector and so I cannot edit its CSS.

**Comment by piroor** [comment]
@confluence You'll need to use `about:debugging` to inspect DOM tree of extension options.
https://github.com/piroor/treestyletab/wiki/How-to-inspect-tree-of-tabs#how-to-start-debugger-for-tree-style-tab

**Comment by confluence** [comment]
I installed the 3.8.3.11565 release, and the issue persists. I can view the settings if I open them in a new tab (I was able to obtain the URL by right-clicking in the invisible embedded page to view the page source, and editing that URL).

**Comment by confluence** [comment]
When I uninstalled the development version and reinstalled the signed version, the problem appeared to fix itself. I'm not sure if this could have something to do with the length of the preferences page -- a freshly installed version has all the sections collapsed by default.

Thanks for the debugger tip -- I will try that if the problem reoccurs.

**Comment by irvinm** [comment]
@piroor I think this can be closed.

**Comment by ngirard** [comment]
I'm afraid the problem remains here.

Firefox 89.0.2 / TST latest version / Ubuntu 21.04.

When clicking on the preferences pane, the vertical scrollbar appears fleetingly, then nothing. The pane just stays blank.

**Comment by piroor** [comment]
@ngirard Could you try inspecting what happens in the options page with the remote debugger?

**Comment by ngirard** [comment]
Hey @piroor, would you mind giving a bit more specific instructions ? Thanks !

**Comment by piroor** [comment]
@ngirard For example:

* Does the root element (`html`) have `initialized` class?
  * If not, please try to add `initialized` to its `class` attribute (it is whitespace-separated list).
* Does the root element (`html`) have `successor-tab-support` class?
* Are any error messages reported in the "Console" tab?

**Comment by ngirard** [comment]
The console tab reports:
```
unreachable code after codemirror-colorpicker.js:1472:4 return statement
```
and the referred code is
```javascript
function unique_array(arrays) {
    return arrays;
    var set = {};
    var count = arrays.length;
    var it = null;
    while (count--) { // <------------------
        it = arrays[count];
        set[JSON.stringify(it)] = it;
    }

    return Object.values(set);
}
```

**Comment by ngirard** [comment]
> Does the root element (html') have initialized` class?

The state of affairs is the same as reported in my previous comment: https://github.com/piroor/treestyletab/issues/2888#issuecomment-835453187

The html element of the preferences tab does have a `successor-tab-support` class as shown in my screenshot, but no `initialized` class.

**Comment by ngirard** [comment]
Adding `initialized` to this element's classes gets the contents of the preferences tab to be displayed ; however its fields aren't populated.

**Comment by piroor** [comment]
Thank you for inspection!

By recent changes I did, all errors in the initialization process is recovered with try-catch statements, but there is one exception: `await` for `configs.$loaded`.
https://github.com/piroor/treestyletab/blob/b7c55103c4ed847b69225317b219df1c44a04df8/webextensions/options/init.js#L542-L587
The class of the root element of the options page has `successor-tab-support` but doesn't have `initialized`, this means that the initialization process is still `await`ing configurations are completely loaded.

Could you try `ext+treestyletab:options` from the location bar? It will loads the options page in a regular tab. If `ext+treestyletab:options` in regular tabs works correctly, it may mean that something happens around inter-processes communications around embedded options page on your Ubuntu environment. I saw similar problems around options page of Thunderbird addons on Tb78, and I had to give up embedded options page on those cases.

**Comment by irvinm** [comment]
@ngirard any updates on this?

**Comment by irvinm** [comment]
@piroor maybe this is "stale"?

**Comment by piroor** [comment]
@irvinm Thanks, this was not expired by the bot because I forgot to add "help wanted" label...

**Comment by ngirard** [comment]
Visiting `ext+treestyletab:options` does load the options page. I'm still using the latest version of TST, this time on Firefox 91.0.1.

I spent quite some time trying to duplicate my Firefox profile and try to determine whether this could be caused by another extension, but to no avail.

I'm feeling a bit like a dead weight, being the sole person reporting this problem.

**Comment by piroor** [comment]
> Visiting ext+treestyletab:options does load the options page. I'm still using the latest version of TST, this time on Firefox 91.0.1.

Hmm, it is very odd - this means that something happens only when Firefox loads the options page into a subframe, in other words this looks to be caused by a bug of Firefox itself like an instability around IPC. I'm very sorry but I cannot imagine how to fix it by TST side.

**Comment by ngirard** [comment]
Just a friendly reminder than I'm still affected by this bug. TST latest, Firefox 94.0 and Ubuntu 21.10.

**Comment by Yenya** [comment]
OK, I see this on one (and only one) of my machines, even with a new session, even under different users (as described in #3128). It appeared immediately after I upgraded this machine to Fedora 36. When I open ext+treestyletab:options, I get the "Gah. Your tab just crashed." window. Also, when I try to open a new session, install and enable Treestyletab in it, close Firefox, and start that session again, I also get the "Gah. Your tab just crashed." window.

Odd thing is, I also upgraded two other machines to Fedora 36, and I don't see the problem there, even with existing sessions.
